### PR TITLE
Qblox auto mixer calibration

### PR DESCRIPTION
**Pull request by hamidelmaazouz**
- Minor refactoring
- More convenient milti-level reset
- Auto mixer cal

### CODE REVIEW

File: src/qat/purr/backends/qblox/device.py [hunk 1]
```diff
@@ -120,12 +129,12 @@ def __init__(
...
         name: str = None,
         address: str = None,
-        cfg: Dict = None,
+        dummy_cfg: Dict = None,
     ):
         super().__init__(id_=dev_id or os.environ.get("QBLOX_DEV_ID"))
         self.name = name or os.environ.get("QBLOX_DEV_NAME")
         self.address = address or os.environ.get("QBLOX_DEV_IP")
-        self.cfg = cfg
+        self.dummy_cfg = dummy_cfg
```
Comment (keriksson-rosenqvist) [inline src/qat/purr/backends/qblox/device.py:137]: Why `dummy`?

File: src/qat/purr/backends/qblox/config.py [hunk 2]
```diff
@@ -170,7 +170,7 @@ class QbloxConfig:
 
 
 class QbloxConfigHelper(ABC):
-    def __init__(self, config: QbloxConfig):
+    def __init__(self, config=QbloxConfig()):
```
Comment (keriksson-rosenqvist) [inline src/qat/purr/backends/qblox/config.py:173]: Minor but why not keep the type hint as is but optional and do:
```
def __init__(self, config: QbloxConfig = None):
    self.config = config or QbloxConfig()
```

File: src/qat/purr/backends/qblox/device.py (hunk 1 above)
Comment (hamidelmaazouz) [reply src/qat/purr/backends/qblox/device.py:137]: config for dummy qblox cluster, it's a "mock" cluster qblox provides for testing

**Review by bgsach — APPROVED** [review]
Talked through in person.

**Pull request merged**